ext/standard/tests/class_object/get_declared_enums_basic.phpt

```diff
@@ -0,0 +1,66 @@
+--TEST--
+Test get_declared_enums() function
+--FILE--
+<?php
+
+enum MyUnitEnum {}
+enum MyStringBackedEnum: string {
+    case FOO = 'test';
+}
+enum MyIntBackedEnum: int {
+    case FOO = 1;
+}
+enum MyEnumImplementation implements Countable {
+    public function count(): int {
+        return 0;
+    }
+}
+
+var_dump(get_declared_enums());
+
+foreach (get_declared_enums() as $enum) {
+    if (!enum_exists($enum)) {
+        echo "Error: $enum is not a valid enum.\n";
+    }
+}
+
+echo "Ensure all enums are included.\n";
+
+var_dump(in_array('MyUnitEnum', get_declared_enums()));
+var_dump(in_array('MyStringBackedEnum', get_declared_enums()));
+var_dump(in_array('MyIntBackedEnum', get_declared_enums()));
+var_dump(in_array('MyEnumImplementation', get_declared_enums()));
+
+echo "Ensure interfaces are not included.\n";
+var_dump(in_array('Throwable', get_declared_enums()));
+var_dump(in_array('IntBackedEnum', get_declared_enums()));
+
+echo "Ensure classes are not included.\n";
+var_dump(in_array('stdClass', get_declared_enums()));
```

Comment:
Possibly redundant, but should there also be a test that `trait`s are not included ?

Reply:
php-src doesn't have any built-in traits so I though to ski it. But I went ahead and added a check with a trait declared in the test itself. Please take a look when you can. Thank you.

Reply:
LGTM. Thanks @Ayesh!